src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectFactory.cs

```diff
@@ -382,6 +386,8 @@ internal void RemoveProjectFromTrackingMaps_NoLock(ProjectId projectId)
             return projectUpdateState;
         }, onAfterUpdateAlways: (projectUpdateState) =>
         {
+            // This is called once after the above transformation is successfully applied.
```

Comment:
mention: including in the case where no change happened.  we want that for yadda yadda.